mate-panel/panel-addto.c

```diff
@@ -662,27 +666,37 @@ static void panel_addto_make_application_model(PanelAddtoDialog* dialog)
 	GtkTreeStore* store;
 	MateMenuTree* tree;
 	MateMenuTreeDirectory* root;
+	GError *error = NULL;
 
 	if (dialog->filter_application_model != NULL)
 		return;
 
 	store = gtk_tree_store_new(NUMBER_COLUMNS, G_TYPE_STRING, G_TYPE_STRING, G_TYPE_POINTER, G_TYPE_STRING);
 
-	tree = matemenu_tree_lookup("mate-applications.menu", MATEMENU_TREE_FLAGS_NONE);
-	matemenu_tree_set_sort_key(tree, MATEMENU_TREE_SORT_DISPLAY_NAME);
+	tree = matemenu_tree_new ("mate-applications.menu", MATEMENU_TREE_FLAGS_SORT_DISPLAY_NAME);
+	if (! matemenu_tree_load_sync (tree, &error)) {
+		g_warning("Menu tree loading got error:%s\n", error->message);
+		g_error_free(error);
+		g_object_unref(tree);
+		tree = NULL;
+	}
 
-	if ((root = matemenu_tree_get_root_directory (tree)))
+	if ((root = matemenu_tree_get_root_directory (tree)) != NULL )
 	{
 		panel_addto_make_application_list(&dialog->application_list, root, "mate-applications.menu");
 		panel_addto_populate_application_model(store, NULL, dialog->application_list);
 
 		matemenu_tree_item_unref(root);
 	}
 
-	matemenu_tree_unref(tree);
+	g_object_unref(tree);
 
-	tree = matemenu_tree_lookup("mate-settings.menu", MATEMENU_TREE_FLAGS_NONE);
-	matemenu_tree_set_sort_key(tree, MATEMENU_TREE_SORT_DISPLAY_NAME);
+	if (! matemenu_tree_load_sync (tree, &error)) {
```

Comment:
The `tree` had just been unref'ed, but we try to load from it. Isn't another `matemenu_tree_new` with "mate-settings.menu" as argument missing here?

Reply:
Yes, I lost one line here.